src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,126 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
 public class StringManipulation implements StringManipulationInterface {
+    private String message;
 
     @Override
     public String getString() {
-        return null;
+        return message;
     }
 
     @Override
     public void setString(String string) {
+        message = string;
     }
 
     @Override
     public int count() {
-        return 0;
+        int counting = 0;
+        for (int i = 0; i < message.length(); i++) {
+            if (message.charAt(i) == ' ') {
+                if (counting == 0) {
+                    if (i > 0 && message.charAt(i - 1) != ' ') {
+                        counting = 1;
+                    }
+                }
+                if (i + 1 < message.length() && message.charAt(i + 1) != ' ') {
+                    counting++;
+                }
+            }
+        }
+        return counting;
```

Comment:
Your code looks good, here's another way of doing it:
public int count() {
        if (string == null) {
            return 0;
        }
        String[] words = string.trim().split("\\s+"); // Split the string into words
        return words.length; // Return the number of words
    }